Algebra.java

```diff
@@ -6,62 +6,182 @@
+	    //System.out.println(plus(2,3));   // 2 + 3 = 5
+		//System.out.println(plus(2,-3));     // 2 + (-3) = -1
+		//System.out.println(plus(-2,3));     // (-2) + 3 = 1
+		//System.out.println(plus(-2,-3));       // (-2) + (-3) = -5
+	    //System.out.println(minus(7,2));  // 7 - 2 = 5
+		//System.out.println(minus(7,-2));    // 7 - (-2) = 9
+		//System.out.println(minus(-7,2));    // (-7) - 2 = -9
+		//System.out.println(minus(-7,-2));      // (-7) - (-2) = -5
+ 		//System.out.println(times(3,4));  // 3 * 4 = 12
+		//System.out.println(times(-3,4));   // (-3) * 4 = -12
+		//System.out.println(times(3,-4));   // 3 * (-4) = -12
+		//System.out.println(times(-3,-4));     // (-3) * (-4) = 12
+   		//System.out.println(plus(5,times(4,2)));  // 5 + 4 * 2
+		//System.out.println(plus(0,times(-9,2)));  // 0 + -9 * 2
+		//System.out.println(plus(-1,times(4,-4)));  // -1 + 4 * -4
+   		//System.out.println(pow(5,3));      // 125
+   		//System.out.println(pow(3,5));      // 243
+		//System.out.println(pow(3,0));         // 1
+		//System.out.println(pow(-3,5));        // -243
+	    //System.out.println(pow(0,5));        // 0
+	    //System.out.println(pow(1,5));        // 1
+   		//System.out.println(div(12,3));   // 4    
+   		//System.out.println(div(-36,-6));    // 1  
+   		//System.out.println(div(25,7));   // 3
+	    //System.out.println(div(-6,2));      //-3
+		//System.out.println(div(6,-2));      //-3
+		//System.out.println(div(0,7));    // 0
+		//System.out.println(div(3,1));    // 3
+   		//System.out.println(mod(6,4));       // 25 % 7
+		//System.out.println(mod(6,3));       // 25 % 7
+		//System.out.println(mod(34,8));       // 2
+   		//System.out.println(mod(120,6));      // 120 % 6   
+		//System.out.println(mod(-8,6));
+		//System.out.println(mod(6,-2)); 
+   		//System.out.println(sqrt(0));
+		//System.out.println(sqrt(263169));
    		System.out.println(sqrt(76123));
+		System.out.println(sqrt(9));
+		System.out.println(sqrt(81));
 	}  
 
 	// Returns x1 + x2
 	public static int plus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		if(x2 < 0) {
+			for(int i = 0; i < Math.abs(x2); i++) {
+				x1--;
+			}
+		} else {
+			for(int i = 0; i < x2; i++) {
+				x1++;
+			}
+		}
+		return x1;
 	}
 
 	// Returns x1 - x2
 	public static int minus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		if(x2 < 0) {
+			for(int i = 0; i < Math.abs(x2); i++) {
+				x1++;
+			}
+		} else {
+			for(int i = 0; i < x2; i++) {
+				x1--;
+			}
+		}
+		return x1;
 	}
 
 	// Returns x1 * x2
 	public static int times(int x1, int x2) {
-		// Replace the following statement with your code
+		int mult = x1;
+		if(x1 == 1)
+		return x2;
+		else if(x2 == 1)
+		return x1;
+		else if(x1 == 0 || x2 == 0)
 		return 0;
+		else if(x1 == -1 && x2 == -1)
+		return 1;
+		else if(x1 == -1 && x2 == 1)
+		return -1;
+		else if(x1 == 1 && x2 == -1)
+		return -1;
+		else if(x1 == 1 && x2 == 1)
+		return 1;
```

Comment:
1. indentations
2. too much , last is never entered (falls in first condition).... 
```
// edge case handles
if (x1 == 0 || x2 == 0){
  return 0;
} else if (x1 == 1) {
  return x2;
} else if (x2 == 1) {
  return x1;
} else if (x1 == -1) {
  return -x2;
} else if (x2 == -1) {
  return -x1;
}

```
and still it can be improved..